File: Logic_HandsOn/C++/Problems on bit,File Handing,Pattern etc/program129.cpp
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept one number from user and toggle 7th and 10th bit of that number. 
//		Return modified number.
//
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////


#include<iostream>
using namespace std;

typedef unsigned int UINT;

class Solution
{
	private:

		UINT iNo;
		UINT iMask;
		UINT iResult;

	public:
		Solution(UINT iNo)
		{
			this->iNo = iNo;
			this->iMask = 0x00000240;
                        this->iResult = 0;

		}

		UINT ToggleBit()
		{	
			iResult = iNo ^ iMask;

			return iResult;
			
		}
};	

int main()
{
	UINT iValue=0;
	UINT uRet=0;

	cout<<"Enter number: ";
	cin>>iValue;

	Solution *sObj = new Solution(iValue);
	uRet = sObj->ToggleBit();

	cout<<"Output: "<<uRet<<"\n";

	return 0;

}

/*
OUTPUT:
Enter number: 137
Output: 713

*/

File: Logic_HandsOn/Java/Problems on bit,file handling,String etc/Program129.java
///////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a program which accept one number from user and toggle 7th and 10th bit of that number. 
//		Return modified number.
//
//////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.Scanner;

class Solution
{
	public int iMask,iNo;
	public int iResult;

	Solution(int iNo)
	{
		this.iNo = iNo;
		this.iMask = 0x00000240;
		this.iResult = 0;
	}

	int ToggleBit()
	{

		iResult = iNo ^ iMask;

		return iResult;
	}	
}

class Program4
{
	public static void main(String[] args)
	{
		int iRet=0;

		Scanner sObj = new Scanner(System.in);
		
		System.out.println("Enter number: ");
		int iValue = sObj.nextInt(); 

		Solution sObj1 = new Solution(iValue);
		iRet = sObj1.ToggleBit();

		System.out.println("Output: "+iRet);
	}	
}
/*
OUTPUT:
Enter number: 
137
Output: 713

*/
















